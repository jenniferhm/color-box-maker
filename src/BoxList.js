import React from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import uuid from 'uuid/v4';

class BoxList extends React.Component {
  constructor(props) {
    super(props);
    this.state = { boxes: [] };
    this.addBox = this.addBox.bind(this);
    this.removeBox = this.removeBox.bind(this);
  }

  addBox(box) {
    let newBox = { ...box, id: uuid() };
    this.setState(st => ({
      boxes: [...st.boxes, newBox]
    }));
  }

  removeBox(boxId) {
    this.setState(st => ({
      boxes: st.boxes.filter(box => box.id !== boxId)
    }))
  }

  renderBoxList() {
    return (
      <div>
        {this.state.boxes.map(box => (
          <Box key={box.id} id={box.id}
            height={box.height}
            width={box.width}
            color={box.color} removeBox={this.removeBox}/>
        ))}
      </div>
    );
  }

  render() {
    return (
      <div>
        <NewBoxForm addBox={this.addBox} removeBox={this.removeBox} />
        {this.renderBoxList()}
      </div>
    )
  }
}



export default BoxList;