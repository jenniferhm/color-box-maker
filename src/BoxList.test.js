import React from "react";
import { shallow, mount} from "enzyme";
import toJson from "enzyme-to-json";
import BoxList from './BoxList';

it("renders without crashing", function() {
  shallow(<BoxList />);
});

it("matches snapshot", function() {
  let wrapper = shallow(<BoxList />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it("renders a list of boxes", function() {
  let wrapper = mount(<BoxList />);

  wrapper.setState({boxes: [
    {id: 1, height: "200px", width: "200px", color: "blue", removeBox: wrapper.state.removeBox },
    {id: 2, height: "200px", width: "200px", color: "green", removeBox: wrapper.state.removeBox },
    {id: 3, height: "200px", width: "200px", color: "yellow", removeBox: wrapper.state.removeBox }
  ]});

  expect(wrapper.state().boxes.length).toEqual(3);
  expect(wrapper.state()).toEqual({boxes: [
    {id: 1, height: "200px", width: "200px", color: "blue", removeBox: wrapper.state.removeBox },
    {id: 2, height: "200px", width: "200px", color: "green", removeBox: wrapper.state.removeBox },
    {id: 3, height: "200px", width: "200px", color: "yellow", removeBox: wrapper.state.removeBox }
  ]});
  expect(wrapper.find("div").last().text()).toEqual("X");
});

it("removeBox function should remove a Box", function() {
  let wrapper = mount(<BoxList />);

  wrapper.setState({boxes: [
    {id: 1, height: "200px", width: "200px", color: "blue", removeBox: wrapper.state.removeBox },
    {id: 2, height: "200px", width: "200px", color: "green", removeBox: wrapper.state.removeBox },
    {id: 3, height: "200px", width: "200px", color: "yellow", removeBox: wrapper.state.removeBox }
  ]});

  wrapper.instance().removeBox(2);

  expect(wrapper.state().boxes.length).toEqual(2);
});

it("addBox function should add a Box", function() {
  let wrapper = mount(<BoxList />);

  wrapper.setState({boxes: [
    {id: 1, height: "200px", width: "200px", color: "blue", removeBox: wrapper.state.removeBox },
    {id: 2, height: "200px", width: "200px", color: "green", removeBox: wrapper.state.removeBox }
  ]});

  wrapper.instance().addBox({id: 3, height: "200px", width: "200px", color: "yellow", removeBox: wrapper.state.removeBox });

  expect(wrapper.state().boxes.length).toEqual(3);
  expect(wrapper.state().boxes[2].id).toEqual(expect.any(String));
});