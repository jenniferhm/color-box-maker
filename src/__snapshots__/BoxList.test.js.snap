// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <NewBoxForm
    addBox={[Function]}
  />
  <div />
</div>
`;
